### ISSUE TITLE
Error: Command failed: git config --get remote.origin.url

### ISSUE THREAD

**Issue by DavidGOrtega** [iterative/cml#618]
Non cloud self-hosted runners might suffer from Git dependency.

```sh
Error: Command failed: git config --get remote.origin.url
    at checkExecSyncError (child_process.js:635:11)
    at execSync (child_process.js:671:15)
    at gitRemoteUrl (/usr/lib/node_modules/@dvcorg/cml/src/cml.js:32:15)
    at new CML (/usr/lib/node_modules/@dvcorg/cml/src/cml.js:77:44)
    at run (/usr/lib/node_modules/@dvcorg/cml/bin/cml-send-comment.js:14:15) {
  status: 1,
  signal: null,
  output: [ null, <Buffer >, <Buffer > ],
  pid: 38581,
  stdout: <Buffer >,
  stderr: <Buffer >
}
```

**Comment by 0x2b3bfa0** [comment]
Yes, all the CML commands require the `git` command-line tool to be present in order to infer the repository address; this shouldn't change. However, we should let users invoke commands with `--repo` if the inferrence is not possible.

**Comment by 0x2b3bfa0** [comment]
Possible workarounds:

* Show a descriptive error message when running commands that need to infer the repository address by running the `git` command, so users can decide to install `git` or use the `--repo` option (#628) 
* Use [`$GITHUB_SERVER_URL/$GITHUB_REPOSITORY`](https://docs.github.com/es/actions/reference/environment-variables) on GitHub and [`$CI_PROJECT_URL`](https://docs.gitlab.com/ee/ci/variables/predefined_variables.html) on GitLab to determine the `gitRemoteUrl` without having to rely on external tools. This would not work outside the supported CI/CD environments.
* Use a JavaScript Git implementation as @DavidGOrtega [suggested](https://github.com/iterative/cml/pull/510/commits/a98b3eec9d4199016f23f70280bd305e68c01353).

**Comment by dacbd** [comment]
Barring the merge of https://github.com/iterative/cml/pull/1003 
also with https://cml.dev/doc/ref#--repo being present now I think its safe to close this.

**Comment by casperdcl** [comment]
Do we need to action some things though?

- update some FAQ in the docs?
- improve the CLI error message #606?
- fix & merge #1003?
